proxy/common/pki.py

```diff
@@ -81,87 +56,72 @@ def gen_private_key(
-    with ssl_config(alt_subj_names, extended_key_usage) as (config_path, has_extension):
-        command = [
-            'openssl', 'req', '-new', '-x509', '-sha256',
-            '-days', str(validity_in_days), '-subj', subject,
-            '-passin', 'pass:%s' % private_key_password,
-            '-config', config_path,
-            '-key', private_key_path, '-out', public_key_path
-        ]
-        if has_extension:
-            command.extend([
-                '-extensions', 'PROXY',
-            ])
-        return run_openssl_command(command, timeout)
+    command = [
+        'openssl', 'req', '-new', '-sha256', '-subj', subject,
+        '-key', private_key_path, '-out', public_key_path
+    ]
+    return run_openssl_command(command, timeout)
 
 
-def gen_csr(
-        csr_path: str,
-        key_path: str,
-        password: str,
-        crt_path: str,
+def gen_proxy_public_key(
+        public_key_path: str,
+        private_key_path: str,
+        subject: str,
+        validity_in_days: int = 365,
         timeout: int = 10) -> bool:
-    """Generates a CSR based upon existing certificate and key file."""
+    """For a given private key, generates a corresponding public key."""
     command = [
-        'openssl', 'x509', '-x509toreq',
-        '-passin', 'pass:%s' % password,
-        '-in', crt_path, '-signkey', key_path,
-        '-out', csr_path
+        'openssl', 'req', '-new', '-x509', '-sha256',
+        '-days', str(validity_in_days), '-subj', subject,
+        '-key', private_key_path, '-out', public_key_path
     ]
     return run_openssl_command(command, timeout)
 
 
-def sign_csr(
-        csr_path: str,
-        crt_path: str,
-        ca_key_path: str,
-        ca_key_password: str,
-        ca_crt_path: str,
-        serial: str,
-        alt_subj_names: Optional[List[str]] = None,
-        extended_key_usage: Optional[str] = None,
-        validity_in_days: int = 365,
-        timeout: int = 10) -> bool:
-    """Sign a CSR using CA key and certificate."""
-    with ext_file(alt_subj_names, extended_key_usage) as extension_path:
-        command = [
-            'openssl', 'x509', '-req', '-sha256',
-            '-CA', ca_crt_path,
-            '-CAkey', ca_key_path,
-            '-passin', 'pass:%s' % ca_key_password,
-            '-set_serial', serial,
-            '-days', str(validity_in_days),
-            '-extfile', extension_path,
-            '-in', csr_path,
-            '-out', crt_path,
-        ]
+def gen_web(csr_path: str,
+            signing_key_path: str,
+            key_path: str,
+            crt_path: str,
+            tld: str,
+            serial: str) -> None:
+    """For a given tld, generate an signed cert."""
+    temp_cert = os.path.join(tempfile.gettempdir(), uuid.uuid4().hex)
+    gen_web_public_key(temp_cert, signing_key_path,
+                       f'/C=/ST=/L=/O=/OU=/CN={tld}')
+    sign_csr(csr_path, key_path, crt_path, temp_cert, serial, [tld])
```

Comment:
FYI, if i add a 
```
os.remove(temp_cert)
```
The e2e test will fail : it looks like the temp_cert is already deleted, but I dont know by who. 
It looks like the OS is cleaning the temp dir automatically, so I propose to dont put the os.remove. 
I let u think about to drop it in other part of the code and let the OS manage those temp files.